bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlay.java

```diff
@@ -391,6 +394,25 @@ private void restoreOverlaySettings() {
 		}
 	}
 
+	@SuppressWarnings("nls")
+	private void assignIDs() {
+		replaceToggle.setData(ID_DATA_KEY, "replaceToggle");
+		searchBar.setData(ID_DATA_KEY, "searchInput");
+		searchBackwardButton.setData(ID_DATA_KEY, "searchBackward");
+		searchForwardButton.setData(ID_DATA_KEY, "searchForward");
+		selectAllButton.setData(ID_DATA_KEY, "selectAll");
+		searchInSelectionButton.setData(ID_DATA_KEY, "searchInSelection");
+		wholeWordSearchButton.setData(ID_DATA_KEY, "wholeWordSearch");
+		regexSearchButton.setData(ID_DATA_KEY, "regExSearch");
+		caseSensitiveSearchButton.setData(ID_DATA_KEY, "caseSensitiveSearch");
+
+		if (replaceBarOpen) {
```

Comment:
Careful: since this method is only being called at startup and on open, there might be a case when not every overlay component has an ID assigned;

1) toggle replace off
2) close and re-open the overlay
3) toggle replace on --> replaceBar does not have an ID set 

This method should likely be called from applyOverlayColors or additionally from toggleReplace

Reply:
It is called from `open()` and `openReplaceDialog()`, which covers all three cases.